rfcs/0050-genesis-ledger-export.md

```diff
@@ -0,0 +1,117 @@
+## Summary
+
+This RFC describes the procedure to generate a genesis ledger from a
+running network, using a node connected to that network.
+
+## Motivation
+
+The procedure described here is a part of the hard fork procedure,
+which aims at spawning a new network, being a direct continuation of
+the mainnet (or any other Mina network for that matter). To enable
+this, the ledger of the old network must be exported in some form and
+then fed into the newly created network. Because the new network's
+initial state can be fed into nodes in a configuration file, it makes
+sense to generate that file directly from the old node. Then necessary
+updates can be made to it manually to update various protocol
+constants, and then the new configuration file can be handed over to
+node operators.
+
+## Detailed design
+
+The genesis ledger export is achieved using a GraphQL field named
+`fork_config`. This field, if asked for, contains a new runtime
+configuration, automatically updated with:
+
+* the dump of the current **staged ledger**, which will become the
+genesis ledger for the new network
+* updated values of `Fork_config`, i.e. previous state hash, previous
+blockchain length and previous global slot.
```

Comment:
This should also include epoch seeds and both the staking ledgers (current and next) if they are different from the genesis ledger of the current network

Reply:
Does the following addition capture the idea?
```suggestion
blockchain length and previous global slot.
* updated epoch data, in particular current and next epoch ledger and seed.
```

Reply:
Mostly. There is this general notion of "epoch data" which is both the ledger as well as some metadata for that ledger (such as total currency). We will need all of that data, along with the epoch seed (which is stored in the protocol state).